server/api/itemRouter.js

```diff
@@ -0,0 +1,115 @@
+const Item = require('../db/models/item');
+const Category = require('../db/models/category');
+const express = require('express');
+const router = express.Router();
+
+/**
+ * Add an item
+ * 
+ * req.body: {sourceLink: String,
+ *            highlight: String,
+ *            userId: mongoose.ObjectId,
+ *            placesId: id(optional),
+ *            comment: String (optional),}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.post('/addItem', (req, res, next) => {
+  const newItem = req.body;
+  const item = new Item(newItem);
+  let response = {}; 
+  item.save((err) => {
+    if (err) {
+      response.success = false;
+      response.message = 'error adding item';
+      res.json(response);
+      next(err);
+    } else {
+      response.success = true;
+      response.message = 'item added';
+      res.json(response);
+    }
+  });
+});
+
+/**
+ * Get all items
+ * 
+ * req.body: {userId: mongoose.objectId}
+ * 
+ * response: [{sourceLink: String,
+ *            highlight: String,
+ *            placeId: id(optional),
```

Comment:
ID is also a string type? To specify a field as optional, the goog closure style goes like
placeId: string=

But no strict requirement of adopting the whole documentation style guide.

Reply:
oh I see. Thank you, I'll use that style for this at least.